From: Amir Vadai <amirv@mellanox.com>
Date: Thu, 20 Jun 2013 14:58:10 +0300
Subject: net/mlx_en: Timestamping is not supported in slave mode
Patch-mainline: v3.10
Git-commit: f9bd2d7f6d18d96fa31a657f35290741117c0dd3
References: bug#919382 FATE#317529

Old hypervisors don't mask out timestamp capability for slave. Till slave
support will be added, need to disable capability by slave.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/mellanox/mlx4/main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -682,6 +682,9 @@ static int mlx4_slave_cap(struct mlx4_de
 		dev->caps.cqe_size   = 32;
 	}
 
+	dev->caps.flags2 &= ~MLX4_DEV_CAP_FLAG2_TS;
+	mlx4_warn(dev, "Timestamping is not supported in slave mode.\n");
+
 	slave_adjust_steering_mode(dev, &dev_cap, &hca_param);
 
 	return 0;
